tests/generic_views/test_edit.py

```diff
@@ -394,3 +394,33 @@ def test_delete_without_redirect(self):
         msg = 'No URL to redirect to. Provide a success_url.'
         with self.assertRaisesMessage(ImproperlyConfigured, msg):
             self.client.post('/edit/author/%d/delete/naive/' % self.author.pk)
+
+    def test_delete_with_form_as_post(self):
+        res = self.client.get('/edit/author/%d/delete/form/' % self.author.pk)
+        self.assertEqual(res.status_code, 200)
+        self.assertEqual(res.context['object'], Author.objects.get(pk=self.author.pk))
+        self.assertEqual(res.context['author'], Author.objects.get(pk=self.author.pk))
```

Comment:
Do we need to re-fetch an author?
```suggestion
        self.assertEqual(res.context['object'], self.author)
        self.assertEqual(res.context['author'], self.author)
```